services/biosamples.py

```diff
@@ -0,0 +1,74 @@
+from typing import List
+from biosamples_v4.api import Client as BioSamplesClient
+from biosamples_v4.aap import Client as AapClient
+from biosamples_v4.encoders import SampleEncoder
+from biosamples_v4.models import Attribute, Sample
+from excel.clean import object_has_attribute, value_populated
+
+
+def attribute_name(name: str) -> str:
+    return name.replace('_', ' ')
+
+
+def optional_attribute(sample: dict, attribute: str):
+    return sample[attribute] if object_has_attribute(sample, attribute) else None
+
+
+def make_attribute(name, value) -> Attribute:
+    units = None
+    # ToDo: Refactor to create the Sample at validation-time so the units are accessible from validation_map
+    if name in ['geographic_location_(latitude)', 'geographic_location_(longitude)']:
+        units = 'DD'
+    return Attribute(name=attribute_name(name), value=value, unit=units)
+
+
+def make_attributes(sample: dict) -> List[Attribute]:
+    attributes = []
+    for name, value in sample.items():
+        if value_populated(value):
+            attributes.append(make_attribute(name, value))
+    return attributes
+
+
+def fixup_sample(sample: dict) -> dict:
+    fixed_sample = sample.copy()
+    remove_keys = ['errors', 'schema_erros', 'sample_accession', 'sample_alias', 'sample_title', 'sample_description', 'tax_id',
```

Comment:
These values could go to a config file, too

Reply:
You could but that would make the code much harder to follow.
Unless we implemented some mapping of excel column names to names for BioSamples (or other services).